test.nix

```diff
@@ -1,317 +1,5 @@
-                      inherit (self) path;
-                    }
-                  );
-                  rust = rust_1_41_0;
-                  rustPackages = self.rust.packages.stable;
-                  inherit (self.rustPackages) rustPlatform;
-                }
-              )
-            ];
-          naersk' = pkgs'.callPackage ./default.nix {};
-        in
-          naersk'.buildPackage
-            {
-              src = ./test/simple-dep;
-              CARGO_BUILD_TARGET = "x86_64-unknown-linux-musl";
-
-            };
-    };
-
-  # local tests, that run pretty fast
-  fastTests =
-    rec
-    {
-      readme = pkgs.runCommand "readme-gen" {}
-        ''
-          cat ${./README.tpl.md} > $out
-          ${docparse}/bin/docparse ${./config.nix} >> gen
-          sed -e '/GEN_CONFIGURATION/{r gen' -e 'd}' -i $out
-        '';
-
-      docparse = naersk.buildPackage {
-        root = ./docparse;
-        src = builtins.filterSource (
-          p: t:
-            let
-              p' = pkgs.lib.removePrefix (toString ./docparse + "/") p;
-            in
-              p' == "Cargo.lock" || p' == "Cargo.toml" || p' == "src" || p' == "src/main.rs"
-        ) ./docparse;
-      };
-
-      readme_test = pkgs.runCommand "readme-test" {}
-        ''
-          diff ${./README.md} ${readme}
-          touch $out
-        '';
-
-      # Tests that the builtDependencies derivation can successfully be unpacked
-      # and that it actually contains cargo's output artifacts. If the result is
-      # ever empty, cargo will still succeed in building the top level crate, except
-      # it will need to rebuild all dependencies from scratch, which is wasteful.
-      # See https://github.com/nix-community/naersk/issues/202
-      depsTargetNotEmpty = pkgs.runCommand "depsTargetNotEmpty"
-        { inherit (simple-dep) builtDependencies; }
-        ''
-          for dep in $builtDependencies; do
-            # Make destination directory for unarchiving
-            mkdir dst
-            ${pkgs.zstd}/bin/zstd -d "$dep/target.tar.zst" --stdout | tar -x -C ./dst
-
-            if [ -z "$(ls -A ./dst)" ]; then
-              echo target directory is empty: "$dep"
-              return 1
-            fi
-
-            rm -rf ./dst
-          done
-
-          # Success
-          touch $out
-        '';
-
-      # Same as the test above except checks when the builtDependencies
-      # derivation is not compressed.
-      depsUncompressedTargetNotEmpty = pkgs.runCommand "depsUncompressedTargetNotEmpty"
-        { inherit (simple-dep-no-compress) builtDependencies; }
-        ''
-          for dep in $builtDependencies; do
-            if [ -z "$(ls -A "$dep"/target)" ]; then
-              echo target directory is empty: "$dep"
-              return 1
-            fi
-          done
-
-          # Success
-          touch $out
-        '';
-
-      simple-dep = naersk.buildPackage {
```

Comment:
To avoid ending up with one, large Nix file, those tests were refactored into `test/fast/test-name/default.nix`